synaptic_reconstruction/imod/to_imod.py

```diff
@@ -84,31 +85,58 @@ def convert_segmentation_to_spheres(
         estimate_radius_2d: If true the distance to boundary for determining the centroid and computing
             the radius will be computed only in 2d rather than in 3d. This can lead to better results
             in case of deformation across the depth axis.
+        props: Optional list of regionprops
 
     Returns:
         np.array: the center coordinates
         np.array: the radii
     """
     num_workers = multiprocessing.cpu_count() if num_workers is None else num_workers
-    props = regionprops(segmentation)
+    if props is None:
+        props = regionprops(segmentation)
+    if props is None:
+        props = regionprops(segmentation)
 
     def coords_and_rads(prop):
         seg_id = prop.label
-
         bbox = prop.bbox
-        bb = np.s_[bbox[0]:bbox[3], bbox[1]:bbox[4], bbox[2]:bbox[5]]
-        mask = segmentation[bb] == seg_id
-
-        if estimate_radius_2d:
-            dists = np.array([distance_transform_edt(ma, sampling=resolution[1:]) for ma in mask])
+        
+        # Handle 2D bounding box
+        if len(bbox) == 4:
+            bb = np.s_[bbox[0]:bbox[2], bbox[1]:bbox[3]]
+            mask = segmentation[bb] == seg_id
+            if resolution:
+                dists = distance_transform_edt(mask, sampling=resolution[:2])
+            else:
+                dists = distance_transform_edt(mask)
+            max_coord = np.unravel_index(np.argmax(dists), mask.shape)
+            radius = dists[max_coord] * radius_factor
+
+            offset = np.array(bbox[:2])
+            coord = np.array(max_coord) + offset
```

Comment:
Don't duplicate all this code. Define a variable `ndim` that is set to 2 / 3 depending on the code path and then use it.

Reply:
I think we'll need this to determine the position